lib/Models/GpxCatalogItem.ts

```diff
@@ -0,0 +1,128 @@
+import i18next from "i18next";
+import { computed, runInAction } from "mobx";
+import createGuid from "terriajs-cesium/Source/Core/createGuid";
+import isDefined from "../Core/isDefined";
+import loadText from "../Core/loadText";
+import readText from "../Core/readText";
+import TerriaError from "../Core/TerriaError";
+import AsyncMappableMixin from "../ModelMixins/AsyncMappableMixin";
+import CatalogMemberMixin from "../ModelMixins/CatalogMemberMixin";
+import UrlMixin from "../ModelMixins/UrlMixin";
+import GpxCatalogItemTraits from "../Traits/GpxCatalogItemTraits";
+import CommonStrata from "./CommonStrata";
+import CreateModel from "./CreateModel";
+import GeoJsonCatalogItem from "./GeoJsonCatalogItem";
+import proxyCatalogItemUrl from "./proxyCatalogItemUrl";
+import getFilenameFromUri from "terriajs-cesium/Source/Core/getFilenameFromUri";
+import LoadableStratum from "./LoadableStratum";
+import { BaseModel } from "./Model";
+import StratumOrder from "./StratumOrder";
+const toGeoJSON = require("@mapbox/togeojson");
+
+class GpxStratum extends LoadableStratum(GpxCatalogItemTraits) {
+  static readonly stratumName = "gpxLoadable";
+
+  constructor(private readonly _item: GpxCatalogItem) {
+    super();
+  }
+
+  duplicateLoadableStratum(newModel: BaseModel): this {
+    return new GpxStratum(newModel as GpxCatalogItem) as this;
+  }
+
+  static async load(item: GpxCatalogItem) {
+    return new GpxStratum(item);
+  }
+}
+
+StratumOrder.addDefinitionStratum(GpxStratum.stratumName);
```

Comment:
This stratum doesn't look like it does anything so it can probably be culled. And then below remove the call to 
`GpxStratum.load(this).then(stratum => {`

Reply:
most of this unused code is from trying to use getFilenameFromUri for the name of the item (when data is loaded from URL) but was always getting an error